idl/partition_checksum.idl.hh

```diff
@@ -99,7 +100,7 @@ struct node_ops_cmd_request {
     // Mandatory field, set by all cmds
     node_ops_cmd cmd;
     // Mandatory field, set by all cmds
-    utils::UUID ops_uuid;
+    node_ops_id ops_uuid;
```

Comment:
I'm not familiar enough with the IDL (@gleb-cloudius ?) - is it ok to use some sort of "tagged UUID" defined in another header file (repair/repair.hh) as part of the IDL?

Reply:
@nyh yes, it's ok.
There are plenty of examples in https://github.com/scylladb/scylladb/blob/master/idl/uuid.idl.hh
that were merged lately.

Reply:
If encoding on the wire is the same it is OK.

Reply:
Oh,  I have an idea - @bhalevy pointed me to uuid.idl.hh which defines host_id et al. Perhaps we should also move the node_ops_id there, instead of repair/repair.hh?

Reply:
> Ok. One of the things that worry me is that when the IDL uses "high level" types from various "high-level" header files (in this case repair/repair.hh), there is the risk that one day the implementation of one of these classes will change, and nobody will even think that they can't be changed because they are part of the IDL. But I agree that this is a theoretical, future-looking, risk, and not a risk today.

After da4f0aae3787992771f481d7dd3bccbacb03f218 the idea is to actually include the external header file when the idl code relies on an external definition.

Reply:
> Oh, I have an idea - @bhalevy pointed me to uuid.idl.hh which defines host_id et al. Perhaps we should also move the node_ops_id there, instead of repair/repair.hh?

Actually, there was an opposition to doing so in a previous PR (by @denesb IIRC?) and we agreed that if a tage_uuid is used only by a single idl file, then it should be defined there so not to litter uuid.idl.hh with a bag full of unrelated definitions.

Reply:
But we can have it in idl/partition_checksum.idl.hh, or even a new idl/node_ops_id.idl.hh, no?

Having it undefined in idl exposes it to a change in repair.hh changing idl, and makes the idl headers non-self-contained.

Reply:
yes, of course.
@Deexie IIRC it was drfined there in the previous version.
Did you delete the definition along with the include statement?

Reply:
I have only defined it in repair.hh. Previously in #11597 I put it in schema_fwd.hh but I do not think it makes sense.

Currently I tried to put that using into idl/node_ops_id.idl.hh, but the parser cannot deal with it.
All the similar types in uuid.idl.hh are defined in some .hh file. Would it be ok to do it analogously? It should be outside of repair/, right?

Reply:
In the idl header you should have a definition like the following:

```diff
diff --git a/idl/partition_checksum.idl.hh b/idl/partition_checksum.idl.hh
index 2515a98a6d..1287af7247 100644
--- a/idl/partition_checksum.idl.hh
+++ b/idl/partition_checksum.idl.hh
@@ -96,6 +95,10 @@ enum class node_ops_cmd : uint32_t {
      repair_updater,
 };
 
+class node_ops_id final {
+    utils::UUID uuid();
+}
+
 struct node_ops_cmd_request {
     // Mandatory field, set by all cmds
     node_ops_cmd cmd;
```

It can be standalone, like `streaming::plan_id`
https://github.com/scylladb/scylladb/blob/2c1ef0d2b768a793c284fc68944526179bfd0171/idl/streaming.idl.hh#L15-L17

but then it could in theory get out of sync with the c++ definition.
plan_id is defined in https://github.com/scylladb/scylladb/blob/2c1ef0d2b768a793c284fc68944526179bfd0171/streaming/stream_fwd.hh#L21
that should have included by `idl/streaming.idl.hh`.

For node_ops_id I think we can just define it in a new repair/id.hh along with repair_uniq_id, and include it in the idl header.
Then repair_task.hh can also rely only on it I believe rather than include "repair/repair.hh"

Reply:
https://github.com/scylladb/scylladb/pull/11720